### ISSUE TITLE
Intersight ruby client require is very expensive (8 seconds, 400 MB, 6000 loaded files)

### ISSUE THREAD

**Issue by jrafanie** [xlab-si/intersight-sdk-ruby#13]
I was playing around with i18n on manageiq and noticed one rake task was extremely slow.  Digging into it, I found that a simple require of the [intersight client](https://github.com/ManageIQ/manageiq-providers-cisco_intersight/blob/48b25ac95868467c9d09cedffd1d97d4103d92d2/app/models/manageiq/providers/cisco_intersight/manager_mixin.rb#L1) was taking several seconds.

I wrote a script to recreate it. You may have to `gem install the sys-proctable` gem if you don't have it locally:

```
% ruby -e 'require "sys/proctable"; puts "memory before: #{Sys::ProcTable.ps(:pid => Process.pid).rss}"; puts "loaded features: #{$LOADED_FEATURES.length}"; puts Time.now; puts "requiring intersight_client"; require "intersight_client"; puts "loaded features: #{$LOADED_FEATURES.length}"; puts Time.now; puts "memory before: #{Sys::ProcTable.ps(:pid => Process.pid).rss}"'

memory before: 15732736
loaded features: 73
2022-06-21 15:22:11 -0400
requiring intersight_client
loaded features: 6165
2022-06-21 15:22:19 -0400
memory before: 422789120
```

Basically, it takes me 8 seconds to do: `require "intersight_client"`
Additionally, it consumes over 400 MB of memory and requires nearly 6000 files.

This is because we're doing "eager" requires here:  https://github.com/xlab-si/intersight-sdk-ruby/blob/main/lib/intersight_client.rb#L14-L5947

I know this gem is generated from openapi but we should look to see if there are ways we can either delay load this using `autoload` (be careful if you're using threads) or do all of these requires when initializing the first "client" object, exposing only the `Client.new` on the normal require.

Some gems will have a skeleton require that exposes the basics and when you first create a client or some objects from this gem, you'd then do a `require/all` or similar to avoid doing this painful require until we're in processes that are actually using the client.

**Comment by jrafanie** [comment]
Related to https://github.com/ManageIQ/manageiq-providers-cisco_intersight/issues/66

**Comment by jrafanie** [comment]
> If you're good with an autoload PR they are very easy to do and I can throw up a PR. I didn't know if the code was auto-generated or not,

yeah, it's generated:  https://github.com/xlab-si/intersight-sdk-ruby/blob/45ec426b9d061502166499a755ac119058002972/lib/intersight_client.rb?rgh-link-date=2022-06-21T19%3A49%3A22Z#L4-L9

I would like to have @zitnik @agrare be aware of this as it's unclear if we rely on the open api generator to periodically push updates based on changes in the api declaration.

**Comment by Fryguy** [comment]
I opened #14 to fix this by replacing the require reorder patch with one that uses autoload.
